compiler-cli/src/dependencies.rs

```diff
@@ -512,6 +514,18 @@ fn get_manifest<Telem: Telemetry>(
     }
 }
 
+// Retrieve dependencies for a given package, and for all local packages it depends on
+fn get_all_dependencies(config: &PackageConfig) -> Result<Dependencies> {
+    let mut dependencies = config.all_dependencies()?;
+    for (_name, requirement) in dependencies.clone().into_iter() {
+        if let Requirement::Path { path } = requirement {
+            let config = crate::config::read(path.join("gleam.toml"))?;
+            dependencies.extend(config.all_dependencies()?);
+        };
+    }
+    Ok(dependencies)
+}
```

Comment:
I think this will fail for nested path deps, as the path will be relative to the dep and not the root of the project.